pkg/model/manifest.go

```diff
@@ -967,7 +968,7 @@ func (m *Manifest) WriteToFile(filePath string) error {
 }
 
 // reorderDocFields orders the manifest to be: name -> build -> deploy -> dependencies -> dev
-func (*Manifest) reorderDocFields(doc yaml3.Node) yaml3.Node {
+func (*Manifest) reorderDocFields(doc *yaml3.Node) *yaml3.Node {
```

Comment:
Why this change?

Reply:
deepsource change

Reply:
I have mixed feelings with this: on one hand `go` doesn't do deep copies, so passing the `doc` by value would not copy the `Content []*Node` which is where all the logic is being done. In this sense, it would be better to pass by *reference. On the other hand, passing by reference would have another behavior (in line `1056` we fill `Content`). If we are getting a pointer we can modify, we should be explicit of *ownership* and change the signature of the function, so that we don't return. 

I think this dichotomy is everywhere, so we should be careful of who *owns* the pointers at any time

Reply:
@mnevadom What's the error? The changes in this PR with pointer de-referencing seem quite dangerous.